packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -16,43 +14,30 @@ const List<String> kUnsupportedVersions = <String>[
 
 /// Regex pattern for identifying line from systeminfo stdout with windows version
 /// (ie. 10.5.4123)
-const String kWindowsOSVersionSemVerPattern =
-    r'^(OS )([^:]*:\s*)([0-9]+\.[0-9]+\.[0-9]+)(.*)$';
+const String kWindowsOSVersionSemVerPattern = r'([0-9]+)\.([0-9]+)\.([0-9\.]+)';
 
 /// Validator for supported Windows host machine operating system version.
 class WindowsVersionValidator extends DoctorValidator {
-  const WindowsVersionValidator({required ProcessManager processManager})
-      : _processManager = processManager,
+  const WindowsVersionValidator({
+    required OperatingSystemUtils operatingSystemUtils,
+  })  : _operatingSystemUtils = operatingSystemUtils,
         super('Windows Version');
 
-  final ProcessManager _processManager;
+  final OperatingSystemUtils _operatingSystemUtils;
 
   @override
   Future<ValidationResult> validate() async {
-    final ProcessResult result =
-        await _processManager.run(<String>['systeminfo']);
-
-    if (result.exitCode != 0) {
-      return const ValidationResult(
-        ValidationType.missing,
-        <ValidationMessage>[],
-        statusInfo: 'Exit status from running `systeminfo` was unsuccessful',
-      );
-    }
-
-    final String resultStdout = result.stdout as String;
-
     final RegExp regex =
         RegExp(kWindowsOSVersionSemVerPattern, multiLine: true);
-    final Iterable<RegExpMatch> matches = regex.allMatches(resultStdout);
+    final Iterable<RegExpMatch> matches =
+        regex.allMatches(_operatingSystemUtils.name);
 
     // Use the string split method to extract the major version
     // and check against the [kUnsupportedVersions] list
     final ValidationType windowsVersionStatus;
     final String statusInfo;
     if (matches.length == 1 &&
-        !kUnsupportedVersions
-            .contains(matches.elementAt(0).group(3)?.split('.').elementAt(0))) {
+        !kUnsupportedVersions.contains(matches.elementAt(0).group(1))) {
       windowsVersionStatus = ValidationType.success;
       statusInfo = 'Installed version of Windows is version 10 or higher';
     } else {
```

Comment:
We should surface what the value of `os.name` was in this case. For example, I can see reports of "The screen cannot be set to the number of lines and columns specified." or "El sistema no puede encontrar el texto del mensaje para el mensaje n£mero 0x2350 en el archivo de mensajes para Application."

Reply:
Maybe:

```dart
    final String name = _operatingSystemUtils.name;
    final Iterable<RegExpMatch> matches =
        regex.allMatches(name);
      
      // etc.
      
      statusInfo = 'Unable to determine Windows version (output from `ver` was "$name")';
```

Reply:
Great suggestion, I've incorporated this in the next push